src/categories.js

```diff
@@ -173,6 +174,23 @@ Categories.getTagWhitelist = function (cids, callback) {
 	db.getSortedSetsMembers(keys, callback);
 };
 
+function calculateUnreadChildren(children, uid) {
+	var cids = children.map(function (child) {
+		return child.cid;
+	});
+
+	Categories.hasReadCategories(cids, uid, function (err, data) {
```

Comment:
This won't work as you expect it to, you are calling an async function inside a synchronous forEach so the execution will just keep going and won't wait for the results of hasReadCategories.

Reply:
calculateUnreadChildren needs a callback parameter as last param and needs to be moved out of the .forEach

Reply:
Worked fine when I tested it. Will fix it in a moment.